specs/receive_funds/01_main.rst

```diff
@@ -0,0 +1,89 @@
++-------+---------------+-----------+------------+
+| issue | title         | author    | created    |
++-------+---------------+-----------+------------+
+| 68_   | Receive funds | tschubotz | 2019-01-17 |
++-------+---------------+-----------+------------+
+
+.. _68: https://github.com/gnosis/safe/issues/68
+
+Receive funds
+===============
+
+1. `Main`_
+2. `User Interface`_
+3. `External Communication`_
+4. Other_
+
+.. _Main:
+
+.. contents:: Table of Contents
+    :depth: 3
+
+1. Problem Definition
+---------------------
+
+- User wants to receive funds
+- User should be able to share their wallet address via different ways
+    - Via QR code for scanning.
+    - Copy to clipboard.
+    - Via other services on their phone.
+- There should be a way to double check if the other person has the
+  correct shared address.
+
+2. Inputs
+-----------
+
+- Address of the currently selected Safe.
```

Comment:
This sounds to me like the user need to input this. For me the input/ requirement is more that a safe needs to exist.

Reply:
Hm ya, I thought specifically on that page, only the Safe address is important, not the rest of the Saf params.  Also e.g. during deployment we know the address, but the Safe doesn't exist, yet.

Reply:
hmm with exist I mean there needs to be a representation of it in the app (also in this case we do require it to exist/ deployment being done, else the UI will be different, if not we should explicitely mention this)

Reply:
isn't this an implementation detail?

in anyway, both versions work for me. We both know what we mean there, so I guess the goal of the spec is reached. ;)

Reply:
it is not.

While the deployment is in progress the designs currently don't display the address. If we want that we need to adjust the designs ;)

Reply:
Maybe to put it like `Address of a safe that would receive the funds`

Reply:
Sorry, but `Address of the currently selected Safe.` sounds exactly like `address of a safe that would receive the funds` to me.

If not the address, what else would be the input?

This really feels to me like we are wasting time discussing this. We all know what is meant, no? So why is `Address of the currently selected Safe.` not a good input? If it's not, what else would be?

Reply:
The only information that this screen needs (to work) is the address of the safe so I agree that the input is the `Address of the Safe`. Being an existent safe or not would then be in the use cases eg.: starting this screen/flow with an invalid Ethereum address **and/or** not being a safe contract **and/or** the app is an owner of the safe.

But for me "selected Safe" already implies that the safe is stored in the device and it exists so that works in my opinion.

Reply:
Yes, that was my thought! The screen technically only needs the address to work. :)

Reply:
But then on the other side, for this feature, the Safe will always exist. So I'll change it to "Address of existing Safe" Let me know if there are objections.